nessie/src/main/java/org/apache/iceberg/nessie/UpdateableReference.java

```diff
@@ -60,13 +60,6 @@ public String getHash() {
     return reference.getHash();
   }
 
-  public Branch getAsBranch() {
```

Comment:
> I deprecated it instead of removing it because it is a public method.

This method a safe cast - you replaced it (not clear for which reason) with an unconditional cast (and BTW left another then unused method).

Reply:
> This method a safe cast

because all the places where this was called, already there is a check for `getRef().checkMutable()`. So, no need for one more method call that checks if it is a branch. Hence, directly casting to branch.

I will remove `isBranch` too.

Reply:
nit: how about returning a `Branch` from `checkMutable` to avoid type casts in the calling code? (just a though, no need to delay this PR because of this).

Reply:
I think `checkMutable` is an early check to ensure we are working on a branch which will later be modified by a commit. Whereas `getRef` or `getAsBranch` is to get the current state of the reference. I am not sure about combining them at the moment. I will explore this in a follow-up a suggested.